models/release.go

```diff
@@ -140,11 +140,35 @@ func createTag(gitRepo *git.Repository, rel *Release) error {
 			return fmt.Errorf("CommitsCount: %v", err)
 		}
 	} else {
+		if !isValidTag(rel.TagName) {
+			return ErrInvalidTagName{rel.TagName}
+		}
 		rel.CreatedUnix = util.TimeStampNow()
 	}
 	return nil
 }
 
+//  See https://git-scm.com/docs/git-check-ref-format for more details
+func isValidTag(tag string) bool {
```

Comment:
We already have ref-name validation: https://github.com/go-gitea/gitea/blob/f3833b7ce4dc78095194808c6e07d8ae133e7ab5/modules/validation/binding.go#L38

I think that `rel.TagName` just needs to be added form validation binding

Reply:
Ugh, I've completely missed that method :sweat: